pkg/project/project.go

```diff
@@ -236,8 +236,17 @@ func (p *Project) CommitID() string {
 }
 
 // CommitUUID returns project commitID in UUID format
-func (p *Project) CommitUUID() strfmt.UUID {
-	return strfmt.UUID(p.commitID)
+func (p *Project) CommitUUID() (strfmt.UUID, error) {
+	if ok := strfmt.Default.Validates("uuid", p.commitID); !ok {
+		return "", locale.NewError("invalid_uuid_val", "Invalid commit ID {{.V0}} in activestate.yaml.  You could replace it with 'latest'", p.commitID)
+	}
+
+	var uuid strfmt.UUID
+	if err := uuid.UnmarshalText([]byte(p.commitID)); err != nil {
+		return "", locale.WrapError(err, "err_commit_id_unmarshal", "Failed to unmarshal the commit id {{.V0}} read from activestate.yaml.", p.commitID)
+	}
+
+	return uuid, nil
```

Comment:
Let's do this parsing in projectfile.Parse(). So this function doesn't need to return any error.